pkg/release/release.go

```diff
@@ -304,91 +305,142 @@ func (r *ReleaseData) parse() error {
 			return fmt.Errorf("failed to copy file out of tar: %w", err)
 		}
 
-		switch {
-		case bytes.Contains(content.Bytes(), []byte("apiVersion: kots.io/v1beta1")):
-			if bytes.Contains(content.Bytes(), []byte("kind: Application")) {
-				parsed, err := parseApplication(content.Bytes())
-				if err != nil {
-					return fmt.Errorf("failed to parse application: %w", err)
-				}
-				r.Application = parsed
-			} else if bytes.Contains(content.Bytes(), []byte("kind: Config")) {
-				parsed, err := parseAppConfig(content.Bytes())
-				if err != nil {
-					return fmt.Errorf("failed to parse app config: %w", err)
-				}
-				r.AppConfig = parsed
+		var documents [][]byte
+		if strings.HasSuffix(header.Name, ".yaml") || strings.HasSuffix(header.Name, ".yml") {
+			// Split multi-document YAML files
+			documents, err = splitYAMLDocuments(content.Bytes())
+			if err != nil {
+				// log only and do not fail here to preserve the previous behavior
+				log.Printf("Failed to parse YAML document from release data %s: %v", header.Name, err)
 			}
+		}
+		if len(documents) == 0 {
```

Comment:
should this be an `else`?

Reply:
I don't think so because if split yaml fails then we add the entire document